api/s2n.h

```diff
@@ -3679,6 +3679,88 @@ S2N_API int s2n_offered_early_data_accept(struct s2n_offered_early_data *early_d
 S2N_API int s2n_config_get_supported_groups(struct s2n_config *config, uint16_t *groups, uint16_t groups_count_max,
         uint16_t *groups_count);
 
+/* Indicates which serialized connection version will be provided. */
+typedef enum {
+    S2N_SERIALIZED_CONN_NONE = 0,
```

Comment:
Nit: might be nice to document the "CONN_NONE" value since it's a little confusing at first glance